CARBON-15650: Migrated carbon configuration from carbon.xml to carbon.yml in C5

dinushab opened the pull request:
- Added carbon.yml to provide configuration instead of carbon.xml
- Added org.wso2.orbit.org.yaml orbit bundle as a dependency for the carbon-kernel
- Added org.yaml.snakeyaml as an import package for carbon core OSGi bundle
- Updated OSGi test cases to be compatible with configuration changes.

Important:
- Old configuration files and obsolete java class have not been removed.
- Before merge this PR, 
  https://github.com/wso2/orbit/pull/149 needs to be merged and org.wso2.orbit.org.yaml:snakeyaml should be published into nexus. Otherwise merging this PR will result in a build break carbon-kernel.
- Review the carbon.yml structure and inform if anything needs to change.

dinushab committed:
CARBON-15650: Added carboy.yml config file and updated bin.xml accordingly.

dinushab committed:
CARBON-15650: Added carboy.yml as a constant in Constants.java

dinushab committed:
CARBON-15650: Added getCarbonYAMLLocation method in Utils.java to return the location of the yaml config file.

dinushab committed:
CARBON-15650: Added YAMLBasedConfigProvider.java and related dependencies into pom files.

dinushab committed:
CARBON-15650:
- Bundling snakeYAML orbit as a runtime feature.
- Importing snakeYAML orbit into Core bundle
- Updating CarbonCoreDSComponent.java to use yaml configuration instead of xml configuration.

dinushab committed:
CARBON-15650: Updated OSGi test cases to work for yaml configuration.

dinushab committed:
CARBON-15650:
- Removed commented code.
- Added new line at the end of carbon.yml

dinushab committed:
CARBON-15650: Refactored javadocs and removed blank lines.

dinushab committed:
CARBON-15650: Removed the blank line above the property "org.snakeyaml.version"

dinushab committed:
CARBON-15650: removed path creating using toString method  and added resolve method in path API.

sameerajayasoma commented on core/pom.xml:
```diff
@@ -132,6 +136,7 @@
                             org.osgi.service.cm.*; version="${osgi.services.cm.package.import.version.range}",
                             org.osgi.service.*;version="${equinox.osgi.services.package.import.version.range}",
                             org.osgi.util.tracker; version="${osgi.service.tracker.package.import.version.range}",
+                            org.yaml.snakeyaml.*;version="${org.snakeyaml.version}"
```
Put a Import package version range

dinushab committed:
CARBON-15650:
- Added version range property in root pom.xml and refering it in pom.xml of carbon core in Import-bundle header.
- Removed StreamReader dependency in YAMLBasedConfigProvider.java and using FileInputStream.
